src/main/java/com/amihaiemil/zold/RtWallet.java

```diff
@@ -84,11 +84,26 @@ public String getId() throws IOException {
     }
 
     @Override
-    public double balance() {
-        throw new UnsupportedOperationException(
-            "Not yet implemented. If you can contribute please, do it at "
-            + "https://github.com/amihaiemil/zold-java-client"
+    public double balance() throws IOException{
+        final HttpGet walletBalance = new HttpGet(
+            URI.create(this.baseUri.toString() + "/balance")
         );
+        try {
+            double value = Double.parseDouble(
```

Comment:
@ammaratef45 , you can just return here, the `value` variable is redundant.

Reply:
@denis-konovalyenko 
You are right
Solved it